department-of-veterans-affairs/vets-website
# MHV-48179 Implement-Front-End-Data-Gap-Fixes-For-Vaccines

## Summary
Updated Vaccines.json with embedded components to address data gaps. Introduced functions extractLocation, extractReaction, and extractNote to extract and organize data for rendering on the VaccineDetails page via the convertVaccine function.

## Related issue(s)
[MHV-48179](https://jira.devops.va.gov/browse/MHV-48179) - Implement front end data gap fixes for Vaccines

User Story: As a Medical Records Team, I want to Implement front end data gap fixes for Vaccines, so that the user has all data available and labeled correctly.

GIVEN:
WHEN:
THEN:

Feature Flag Y/N ? N
Manual Testing Y/N ? Y
Automated Testing Y/N ? Y
Accessibility Testing Y/N ? N

Acceptance Criteria:
AC1 Make sure all the fields listed in the comments are mapped appropriately
AC2 Add "Manufacturer" field to Web, Print, and PDF, also mapped appropriately (see Sketch)
AC3 Don't access the contained[] resources directly – do a lookup via performer/reaction respectively (see allergy for reference).
AC4
AC5

Architectural Notes:

Sketch: https://www.sketch.com/s/4c2728ff-649d-4212-98d2-04c9b3fff9d4/a/4o55JJq

This PR is the corresponding work vitals, and might help tackle this issue: https://github.com/department-of-veterans-affairs/vets-website/pull/26289

## Testing done

Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots

Note:  The entries Health1, VaxGuard, ShotSafe, and StayWell were included solely for testing purposes, demonstrating how the array behaves when containing more than one element. The actual output will include "test comment."

![Screenshot 2023-11-27 at 9 44 08 AM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/69a17c10-d8c2-415b-89c8-72798cf256c3)

![Screenshot 2023-11-27 at 9 59 29 AM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/4a06aa65-f150-4106-9e38-593111c2fa1b)



## What areas of the site does it impact?

The modification affects the vaccineDetail section of the Medical Records page.

### Quality Assurance & Testing


### Authentication
